src/main/java/com/fivetran/agent/mysql/output/BucketOutput.java

```diff
@@ -133,7 +133,7 @@ private void flushToBucket() throws IOException {
         refreshFileChannel();
 
         Path statePath = Paths.get("mysql_state.json");
-        FileChannel stateFileChannel = FileChannel.open(statePath, StandardOpenOption.CREATE_NEW, StandardOpenOption.WRITE);
+        FileChannel stateFileChannel = FileChannel.open(statePath, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.WRITE);
```

Comment:
I'm not sure this is actually necessary. I changed it because tests were doing some bad behavior, but in another PR that fixes the test, I think I came up with a true fix for the bad behavior (and it wasn't this).

Reply:
We'll have to merge the test fix branch into this one to find out.